### PR TITLE
JC19-381-search

### PR DESCRIPTION
**Pull request by daoduong-saritasa**
Search feature

### CODE REVIEW

**Comment by panov-aleksey-saritasa** [comment]
@daoduong-saritasa please, fix merge conflicts, then close and reopen this pr, it should trigger invites action

File: apps/vanilla/src/namespaces/PageHandler.ts [hunk 1]
```diff
@@ -12,10 +12,13 @@ export namespace PageHandler {
    * @param newPage Next page number.
    */
   export function goToPageByNum(newPage: number): void {
+    const searchQuery = localStorage.getItem(SEARCH_QUERY);
```
Comment (artem-sotnichenko-saritasa) [inline apps/vanilla/src/namespaces/PageHandler.ts:15]: Please be consistent. As I can see, in this code constants for local storage use `LS` suffix.

File: apps/vanilla/src/namespaces/SearchHandler.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+import { OFFSET, FIRST_PAGE, LIMIT, SEARCH_QUERY, SORT_LS } from '../script/constants';
+import { updateTable } from '../services/fetchAnime';
+import { assertNonNullish } from '../utils/assertNonNullish';
+
+export namespace SearchHandler {
+
+  /** Handle searching functionality. */
+  export function handleSearch(): void {
+    const searchText = document.querySelector<HTMLInputElement>('.search__input');
```
Comment (artem-sotnichenko-saritasa) [inline apps/vanilla/src/namespaces/SearchHandler.ts:11]: The variable name is misleading.
You have element assigned into variable, but name suggests there's a string.

**Review by artem-sotnichenko-saritasa — APPROVED** [review]

**Review by kristinaSvobodinaSaritasa — APPROVED** [review]

File: apps/vanilla/src/script/localStorageName.ts [hunk 3]
```diff
@@ -2,3 +2,4 @@
 export const COUNT_LS = 'COUNT';
 export const SORT_QUERY = 'ordering';
 export const PAGE_QUERY = 'page';
+export const SEARCH_LS = 'SEARCH';
```
Comment (timofei-maximov-saritasa) [inline apps/vanilla/src/script/localStorageName.ts:5]: What does `LS` stands for? Please, improve naming

**Review by maxineo — APPROVED** [review]

**Pull request merged**